### PR TITLE
Remove diskstate from vdslib.

### PR DESCRIPTION
**Pull request by toregge**
@vekterli : please review
@geirst , @baldersheim : FYI

### CODE REVIEW

File: storage/src/tests/distributor/bucketdbupdatertest.cpp [hunk 1]
```diff
@@ -1474,10 +1474,6 @@ TEST_F(BucketDBUpdaterTest, pending_cluster_state_send_messages) {
               getSentNodes("distributor:10 storage:2",
                            "distributor:10 .1.s:d storage:2"));
 
-    EXPECT_EQ(getNodeList({1}),
-              getSentNodes("distributor:2 storage:2",
-                           "distributor:2 storage:2 .1.d:3 .1.d.1.s:d"));
-
```
Comment (vekterli) [inline storage/src/tests/distributor/bucketdbupdatertest.cpp:1480]: Should the test assertion below it be removed as well? It also mentions an explicit disk state

**Review by baldersheim — APPROVED** [review]
👍

File: storage/src/vespa/storage/storageserver/servicelayernode.cpp [hunk 2]
```diff
@@ -105,20 +105,9 @@ ServiceLayerNode::initializeNodeSpecific()
...
-        if (i >= ns.getDiskCount()) {
-            LOG(warning, "Capacity configured for partition %zu but only %u partitions found.",
-                _serverConfig->diskCapacity.size(), ns.getDiskCount());
-            continue;
-        }
-        lib::DiskState ds(ns.getDiskState(i));
-        ds.setCapacity(_serverConfig->diskCapacity[i]);
-        ns.setDiskState(i, ds);
-    }
     LOG(debug, "Adjusting reported node state to include partition count and states, capacity and reliability: %s",
```
Comment (vekterli) [inline storage/src/vespa/storage/storageserver/servicelayernode.cpp:111]: Consider updating log message to no longer mention partition

**Review by vekterli — APPROVED** [review]
👍🎉

File: storage/src/tests/distributor/bucketdbupdatertest.cpp (hunk 1 above)
Comment (toregge) [reply storage/src/tests/distributor/bucketdbupdatertest.cpp:1480]: Removed.

File: storage/src/vespa/storage/storageserver/servicelayernode.cpp (hunk 2 above)
Comment (toregge) [reply storage/src/vespa/storage/storageserver/servicelayernode.cpp]: Fixed.

**Pull request merged**